### PR TITLE
frontend: created "Receive" btn for empty wallet CTA

### PR DESCRIPTION
**Pull request by shonsirsha**
We'd like to have a "receive" btn (instead of only "buy") in the CTA as it's also a valid way of depositing coins.

The following points are the outlines of this PR:

1. Modified the buyCTA to also able to include "Receive" btn. This is displayed in the account summary screen.

2. Extracted the account selector into a single component so we can reuse it in the new receive screen and still use it in the buy screen.

3. Created a new receive screen/page (`/select-receive`) which displays the said selector.

4. If there's only one account, upon clicking the "Receive" btn user won't be broguht to `/select-receive`, but instead directly to their only account's receive screen.

5. Re-designed empty account UI.

[Preview] Only has BTC accounts:

https://user-images.githubusercontent.com/28676406/212833909-09fc1cf8-9a0f-454e-a587-beb91541f031.mov

[Preview] Has BTC and LTC accounts (not only BTC):

https://user-images.githubusercontent.com/28676406/212833796-3673ba09-7690-4b59-82bb-1673f809e036.mov

Updated Preview:
Screenshot 2023-01-30 at 07 38 25

Empty account:

Account Selector 
Screenshot 2023-01-30 at 07 34 13

The account selector container / layout is now aligned to the top because we're using `View`, and `View` doesn't support centre alignment yet. Will make a separate PR to add this functionality.

### CODE REVIEW

File: frontends/web/src/routes/receive/receive.tsx
Comment (shonsirsha) [inline frontends/web/src/routes/receive/receive.tsx:55]: One thing, I included a Guide here. I took it from the individual receive page (`accounts/${code}/receive`). 

It can (probably should) be changed later on, just to make sure we have guide included here.

Feel free to lmk if we have Guide content ready for this new receive page, and I can put them here. :)

Comment (thisconnect) [reply frontends/web/src/routes/receive/receive.tsx:55]: No guide needed (for now=.

File: frontends/web/src/routes/account/info/buyReceiveCTA.tsx
Comment (thisconnect) [inline frontends/web/src/routes/account/info/buyReceiveCTA.tsx:38]: I find 2 components and 2 routes with the name receive a bit confusing.

how about `src/routes/account/receive/receive-select.tsx` and `account/receive-select` or similar?

Screen Shot 2023-01-25 at 16 04 24

File: frontends/web/src/components/accountSelector/accountselector.tsx
Comment (thisconnect) [inline frontends/web/src/components/accountSelector/accountselector.tsx:38]: I am not sure about the title repeating here. lets discusss dm

File: frontends/web/src/components/accountSelector/accountselector.module.css
Comment (thisconnect) [inline frontends/web/src/components/accountSelector/accountselector.module.css:5]: there is also an empty file `frontends/web/src/routes/receive/receive.module.css` but doesn't let me comment there, remove that?

File: frontends/web/src/components/accountSelector/accountselector.tsx
Comment (thisconnect) [reply frontends/web/src/components/accountSelector/accountselector.tsx:38]: instead of using a new custom title could you use `class="subTitle"` (should be centered if you use `<View>` component lets check together tomorrow maybe or next week).

File: frontends/web/src/routes/receive/receive.tsx
Comment (shonsirsha) [reply frontends/web/src/routes/receive/receive.tsx:47]: Yeah, right we've talked about this. Will replace this using `View`.

File: frontends/web/src/components/accountSelector/accountselector.module.css
Comment (shonsirsha) [reply frontends/web/src/components/accountSelector/accountselector.module.css:5]: oops.. thats quite random. will remove it!

File: frontends/web/src/routes/account/info/buyReceiveCTA.tsx
Comment (shonsirsha) [reply frontends/web/src/routes/account/info/buyReceiveCTA.tsx:38]: Yeah I agree it's a bit confusing.

 What about `receive-select` for the new page w/ select and `account-receive` for the current receive page (the one for each account)?

Comment (thisconnect) [reply frontends/web/src/routes/account/info/buyReceiveCTA.tsx:38]: I think the file and the route for recieve.tsx  should stay as is, 

> What about receive-select for the new page w/ select and account-receive for the current receive page (the one for each account)?

for route or filename? I think both route and filename should use the same naming.

Comment (shonsirsha) [reply frontends/web/src/routes/account/info/buyReceiveCTA.tsx:38]: Ok. Lets just use /receive-select for the new page orrr /accounts-receive. Then change the route accordingly.

File: frontends/web/src/components/accountSelector/accountselector.tsx
Comment (shonsirsha) [reply frontends/web/src/components/accountSelector/accountselector.tsx:38]: `.subtitle` has a different styling (heavier font): 

Screenshot 2023-01-26 at 07 50 51

intentional (do we wanna use it here)?

Comment (shonsirsha) [reply frontends/web/src/components/accountSelector/accountselector.tsx:38]: looks like `subtitle` has a different style (smaller font):

Screenshot 2023-01-26 at 07 52 05
intentional / do we actually wanna use smaller font here?   

Moreover, the `View`  component doesn't have sufficient styling to achieve what we want in this page. We need it vertically centred and has a narrow size (max-width to be of certain size, check `.content.narrow`) - and `View` has neither of these options. 

Since we plan to use View everywhere and dropping the container divs, shall I work on `View` first to allow for what we need, and then use it here?

Comment (thisconnect) [reply frontends/web/src/components/accountSelector/accountselector.tsx:38]: subTitle is used in many components. in the same file (layout.css) there is also `.title` you can try that too, but I don't see where it is used in the codebase, hard to search for.

Comment (thisconnect) [reply frontends/web/src/components/accountSelector/accountselector.tsx:38]: I think `subTitle` is fine, it gets bigger on large screen. also ViewHeader uses the same font size through CSS variable, `var(--size-subheader)`. please have a look at passphrase.tsx (under device-settings go to enable passphrase, there is some titles)

Comment (shonsirsha) [reply frontends/web/src/components/accountSelector/accountselector.tsx:38]: Yes, `.title` is the exact class for this I think :)

**Comment by shonsirsha** [comment]
Thanks a lot for the reviews and discussion @thisconnect ! Made some changes according to our communications. PTAL 🙏

File: frontends/web/src/routes/account/info/buyReceiveCTA.tsx
Comment (thisconnect) [inline frontends/web/src/routes/account/info/buyReceiveCTA.tsx:44]: I can't find `columns-container` class, maybe `styles.main` is enough?

**Review by thisconnect — CHANGES_REQUESTED** [review]
nice with this change "empty accounts" look much cleaner.

- we probably should merge the centeredVertical first as this could benefit from it (added comment)
- this might conflict with `staging-pocket` which is ready soon, prob wait to get pocket merged first? 
- there is another account select in request address workflow, here is a test commit to trigger this view https://github.com/shonsirsha/bitbox-wallet-app/pull/1 maybe the accountselector could be used there as well.

File: frontends/web/src/routes/account/info/buyReceiveCTA.tsx
Comment (shonsirsha) [reply frontends/web/src/routes/account/info/buyReceiveCTA.tsx:44]: Yes, seems  like it's an old CSS. styles.main is enough.

File: frontends/web/src/routes/account/info/buyReceiveCTA.module.css
Comment (shonsirsha) [reply frontends/web/src/routes/account/info/buyReceiveCTA.module.css:15]: unsure either, again old CSS. I imagine this is just setting a default font-weight (which is 400). I'll see if we can remove this, which I think is possible anyway.

File: frontends/web/src/components/accountSelector/accountselector.tsx
Comment (shonsirsha) [reply frontends/web/src/components/accountSelector/accountselector.tsx:36]: Yeah.

**Comment by shonsirsha** [comment]
Made some more changes, @thisconnect  PTAL 🙏 

>we probably should merge the centeredVertical first as this could benefit from it (added comment)

Agree 👍 

> this might conflict with staging-pocket which is ready soon, prob wait to get pocket merged first?

Also agree, as that branch is bigger I could resolve the conflicts here more easily. :) That one is also relatively more crucial / urgent to be merged.

To sum-up, will merge this once staging-pocket and #1935 are merged.

File: frontends/web/src/routes/account/account.tsx
Comment (thisconnect) [inline frontends/web/src/routes/account/account.tsx:192]: nit: it would help me for readability when the `isAccountEmpty` with many conditions would keep using newlines.

and the `showBuyButton ` with only `moonpayBuySupported && isAccountEmpty ` is fine on the same line.

Comment (thisconnect) [reply frontends/web/src/routes/account/account.tsx:192]: I assume with `staging-pocket` merged, there needs to be also a pocketBuySupprted similar to `moonpayBuySupported`.

File: frontends/web/src/routes/accounts/select-receive.tsx
Comment (thisconnect) [inline frontends/web/src/routes/accounts/select-receive.tsx:47]: once the View PR is merged this should use `verticallyCentered`.

**Review by thisconnect — COMMENTED** [review]
LGTM but onhold + with small nit and this PR should be updated once pocket is in.

File: frontends/web/src/routes/account/info/buyReceiveCTA.module.css
Comment (shonsirsha) [inline frontends/web/src/routes/account/info/buyReceiveCTA.module.css:7]: Made a small change on the css  adding a `:first-child` selector to fix the following margin issue:

Before:
Screenshot 2023-02-13 at 07 08 20


After:

Screenshot 2023-02-13 at 07 08 10

File: frontends/web/src/routes/accounts/select-receive.tsx
Comment (thisconnect) [reply frontends/web/src/routes/accounts/select-receive.tsx:47]: @shonsirsha https://github.com/digitalbitbox/bitbox-wallet-app/pull/1935 was merged you can use verticallyCentered now.

Comment (shonsirsha) [reply frontends/web/src/routes/accounts/select-receive.tsx]: Done! 🙏

Comment (shonsirsha) [reply frontends/web/src/routes/accounts/select-receive.tsx]: @thisconnect done 🙏

**Comment by thisconnect** [comment]
please rebase

**Comment by shonsirsha** [comment]
@thisconnect rebased. Please feel free to test and take a look once more. Thanks 🙏

File: frontends/web/src/components/view/view.module.css
Comment (shonsirsha) [inline frontends/web/src/components/view/view.module.css:83]: Had to add this in order to mitigate this bug (I think we missed it):

**Before,** when using `View` with `verticallyCentered` on mobile, it doesn't vertically center:

Screenshot 2023-02-21 at 10 15 32

**After**
Screenshot 2023-02-21 at 10 15 43

File: frontends/web/src/routes/account/info/buyReceiveCTA.module.css
Comment (thisconnect) [reply frontends/web/src/routes/account/info/buyReceiveCTA.module.css:7]: tiny nit: the line-height of subTitle looks a bit too small, but this class is used in different places as well so lets keep as is and look at it in a different PR. I am not sure if other components depend on `line-height: 1;` or if that is ok to change globally.

File: frontends/web/src/components/view/view.module.css
Comment (thisconnect) [reply frontends/web/src/components/view/view.module.css:83]: yes View still is not perfect and needs more attention. I looked into that as well yesterday and made a similar change https://github.com/digitalbitbox/bitbox-wallet-app/commit/b1079769e23bf01df574bd8165941b6033fa7802

File: frontends/web/src/components/accountSelector/accountselector.tsx
Comment (thisconnect) [inline frontends/web/src/components/accountSelector/accountselector.tsx:63]: This component shouldn't use View only routes should.
also there is ViewButtons, I am thinking if it would make sense to export AccountSelector and AccountSelectorButton separatelly so in the view it could be placed in different containers.

**Review by thisconnect — CHANGES_REQUESTED** [review]

File: frontends/web/src/components/accountSelector/accountselector.tsx
Comment (shonsirsha) [reply frontends/web/src/components/accountSelector/accountselector.tsx:63]: Oops, yeah!

**Comment by shonsirsha** [comment]
Thanks @thisconnect . Made the changes, PTAL :)

**Review by thisconnect — CHANGES_REQUESTED** [review]

**Review by thisconnect — COMMENTED** [review]
found 2 small issues, sat instead of crypto bug is unrelated to this PR, up to you if you want to fix it in this PR.

But the translation would be nice to have fixed.

**Comment by shonsirsha** [comment]
@thisconnect Made the changes for translation (also noticed one more occurrence in `select-receive` so fixed it there too). The sat instead of crypto bug thing also gets fixed in this PR. :)

**Review by thisconnect — APPROVED** [review]
thanks 👍 tested LGTM!

tested with servewallet-mainnet and qt-osx

**Pull request merged**
